metricbeat/module/logstash/node_stats/data_xpack_test.go

```diff
@@ -0,0 +1,52 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build !integration
+
+package node_stats
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMakeClusterToPipelinesMap(t *testing.T) {
+	pipelines := []PipelineStats{
+		{
+			ID: "test_pipeline",
+			Vertices: []map[string]interface{}{
+				{
+					"id": "vertex_1",
+				},
+				{
+					"id": "vertex_2",
+				},
+				{
+					"id": "vertex_3",
+				},
+			},
+		},
+	}
```

Comment:
Is this particular case able to reproduce the issue we have had? If not, we should one.

The expannsion into the cluster map is somewhat critical in order to produce correct docs, that show the correct association of configurations to single Elasticsearch clusters. Can we make this test more exhaustive?

Reply:
Yes, this is the case that was problematic. Before this PR this case would've produced 3 `PipelineState` associated with the `prod_cluster_id` cluster UUID when we should've only got 1 `PipelineState`.

I will expand this test to add a few more test cases.

Reply:
Added more test cases in 672ee5e05.

Reply:
Thank you!